dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTablesIntegrationTest.java

```diff
@@ -0,0 +1,208 @@
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.dotcms.util.CollectionsUtils.list;
+import static com.dotcms.util.CollectionsUtils.map;
+import static org.junit.Assert.assertEquals;
+
+
+public class Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTablesIntegrationTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to Test: {@link Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables#executeUpgrade()}
+     * When: Run the Upgrade Task
+     * Should: Should remove the foreign key to all the Integrity Resolver tables and rename the column endpoint_id
+     * The Integrity Resolver tables are:
+     * - cms_roles_ir
+     * - folders_ir
+     * - structures_ir
+     * - htmlpages_ir
+     * - fileassets_ir
+     *
+     * After remove the constraint you should be allow to insert register in this table without has any register in publishing_end_point
+     */
+    @Test
+    public void constraintShouldNotExists() throws DotDataException, SQLException {
+
+        final String endpointId = insertPublishingEndPoint();
+        insertFolderIntegrityResolver(endpointId);
+        insertPageIntegrityResolver(endpointId);
+        insertFileAssetIntegrityResolver(endpointId);
+        insertRolesIntegrityResolver(endpointId);
+        insertStructuresIntegrityResolver(endpointId);
+
+        final Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables =
+                new Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables();
+
+        task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables.executeUpgrade();
+
+        insertFolderIntegrityResolver("anyIP");
+        insertPageIntegrityResolver("anyIP");
+        insertFileAssetIntegrityResolver("anyIP");
+        insertRolesIntegrityResolver("anyIP");
+        insertStructuresIntegrityResolver("anyIP");
+
+        checkColumnsSize();
+    }
+
+    private void checkColumnsSize() throws SQLException {
+        final List<String> tables = list(
+                "folders_ir",
+                "structures_ir",
+                "htmlpages_ir",
+                "fileassets_ir",
+                "cms_roles_ir"
+        );
+
+        for (String table : tables) {
+            checkColumnSize(table);
+        }
+    }
+
+    private void checkColumnSize(final String tableName) throws SQLException {
+        final Connection connection = DbConnectionFactory.getConnection();
+        final ResultSet resultSet = DotDatabaseMetaData.getColumnsMetaData(connection, tableName);
```

Comment:
Codacy found an issue: [Ensure that resources like this ResultSet object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)